src/helpers/misc.ts

```diff
@@ -131,13 +131,22 @@ export function getDefaultCellHeight(style: Style | undefined, numberOfLines?: n
 }
 
 export function computeTextWidth(context: CanvasRenderingContext2D, text: string, style: Style) {
-  context.save();
-  context.font = computeTextFont(style);
-  const textWidth = context.measureText(text).width;
-  context.restore();
-  return textWidth;
+  const font = computeTextFont(style);
+  if (!textWidthCache[font]) {
+    textWidthCache[font] = {};
+  }
+  if (textWidthCache[font][text] === undefined) {
+    context.save();
+    context.font = font;
+    const textWidth = context.measureText(text).width;
+    context.restore();
+    textWidthCache[font][text] = textWidth;
+  }
+  return textWidthCache[font][text];
 }
 
+const textWidthCache: Record<string, Record<string, number>> = {};
```

Comment:
I copied Odoo's memoize helper for my locale PR, you can maybe use it here 

https://github.com/odoo/o-spreadsheet/pull/2483/commits/63910bc83eae201b583b9a89e35311fc8ebcc0d8#diff-95a28887255e747473f4dfa92952fcded1631165f722a3e898685e2b8ebffcf7R568

Reply:
It only works with a single argument. It can be used to memoize a function with multiple arguments but bon...
```js
function fn(a, b) { ... }
const fnMemo = memoize(a => memoize(b => fn(a, b)))
fnMemo(1)(2);
// or uncurried
const fnMemoUncurried = (a, b) => memoize(a => memoize(b => fn(a, b)))(a)(b)
fnMemoUncurried(1, 2)
```
...it quickly looks hairy
So I have mixed feelings about that

Reply:
as long as we don't need a global solution, we could stick with this